deps/rabbit/src/rabbit_fifo_client.erl

```diff
@@ -863,7 +915,39 @@ handle_delivery(_QName, _Leader, {delivery, Tag, [_ | _] = IdMsgs},
     %% we should return all messages.
     MsgIntIds = [Id || {Id, _} <- IdMsgs],
     {State1, Deliveries} = return(Tag, MsgIntIds, State0),
-    {ok, State1, Deliveries}.
+    {ok, State1, Deliveries};
+handle_delivery(QName, Leader, {delivery, Tag, ReadPlan, Msgs},
+                #state{cached_segments = CachedSegments} = State) ->
+    {TRef, Cached0} = case CachedSegments of
+                          undefined ->
+                              {undefined, undefined};
+                          {R, _, C} ->
+                              {R, C}
+                      end,
+    {MsgIds, Cached1} = rabbit_fifo:exec_read(Cached0, ReadPlan, Msgs),
+    %% if there are cached segments after a read and there
+    %% is no current timer set, set a timer
+    %% send a message to evict cache after some time
+    Cached = case ra_flru:size(Cached1) > 0 of
+                 true when TRef == undefined ->
+                     Ref = erlang:send_after(?CACHE_SEG_TIMEOUT, self(),
+                                             {'$gen_cast',
+                                              {queue_event, QName,
+                                               {Leader, close_cached_segments}}}),
+                     {Ref, now_ms(), Cached1};
+                 true ->
+                     {TRef, now_ms(), Cached1};
+                 false ->
+                     if TRef =/= undefined ->
+                            _ = erlang:cancel_timer(TRef, [{async, true}]),
```

Comment:
Is this line tested?
Given that `Info` defaults to `true`, I would this line expect to crash the channel process from what I read in https://www.erlang.org/doc/apps/erts/erlang.html#cancel_timer/2
> a message on the form {cancel_timer, TimerRef, Result} is sent to the caller of erlang:cancel_timer() when the cancellation operation has been performed

Reply:
no I doubt this line is tested, it would be very hard to write a reliable test for it such that the timer is set but the cache is empty. In fact I'm not 100% sure it can happen unless we can trigger some weird combo of a missing key combined with some entries in the mem table and some not.
 
 I will add `{info, false}` though

Reply:
good catch